src/app/search-page/search.component.ts

```diff
@@ -0,0 +1,226 @@
+
+  selectedReleaseDate: string = '';
+
+  private readonly destroy$ = new Subject<void>();
+
+  private _searchDetails: string = '';
+
+  constructor(private route: ActivatedRoute,
+    private router: Router,
+    private productService: ProductService,
+    private queryService: QueryService,
+    private filterService: FilterService) { }
+
+
+  get searchDetails(): string {
+    return this._searchDetails;
+  }
+
+  set searchDetails(value: string) {
+    this._searchDetails = value;
+  }
+
+  resetFilterValues() {
+    this.selectedStatus = '';
+    this.selectedCategory = '';
+    this.selectedReleaseDate = '';
+  }
+
+  onSearch(): void {
+    this.router.navigate(["search", this._searchDetails]);
+    this.query = this._searchDetails;
+
+    this.productService.getProductsByQuery(this._searchDetails)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe({
+        next: data => {
+          this.resetFilterValues();
+          this.filteredProducts = [];
+          this.allProductsBySearch = data.results;
+          this.searchCount = data.count;
+          this.visibleProducts = this.allProductsBySearch.slice(0, 12);
+          this.getCategoryList();
+          this.getStatusList();
+        }
+      })
+  }
+
+  onPageChange(event: PageEvent, productsArray: IProduct[]) {
+
+    const startIndex = event.pageIndex * event.pageSize;
+    let endIndex = startIndex + event.pageSize;
+
+    if (endIndex > productsArray.length) {
+      endIndex = productsArray.length
+    }
+    this.visibleProducts = productsArray.slice(startIndex, endIndex);
+
+  }
+
+  getCategoryList() {
+    const unfilteredCategories: string[] = [];
+
+    this.allProductsBySearch.map(item => {
+      unfilteredCategories.push(item.category)
+    })
+    const uniqueCategories = findUniqueItems(unfilteredCategories);
+
+    this.categoryList = uniqueCategories;
+  }
+
+  getStatusList() {
+    const unfilteredStatusList: string[] = [];
+
+    this.allProductsBySearch.map(item => {
+      unfilteredStatusList.push(item.status)
+    })
+    const uniqueStatusList = findUniqueItems(unfilteredStatusList);
+
+    this.statusList = uniqueStatusList;
+  }
+
+  ngOnInit(): void {
+    this.query = this.route.snapshot.paramMap.get("query");
+
+    this.filterService.categoryData$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(category => {
+        this.resetFilterValues();
+        this.selectedCategory = category;
+        this.filteredProducts = findItemsByCategory(this.selectedCategory, this.allProductsBySearch);
+        this.visibleProducts = this.filteredProducts.slice(0, 12);
+      })
+
+    this.filterService.statusData$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(status => {
+        this.resetFilterValues();
+        this.selectedStatus = status;
+        this.filteredProducts = findItemsByStatus(this.selectedStatus, this.allProductsBySearch);
+        this.visibleProducts = this.filteredProducts.slice(0, 12);
+      })
```

Comment:
These functions are almost identical.
When this is the case it makes sense to move the identical parts into a new function.